sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -746,6 +749,345 @@ case class DescribeTableCommand(
   }
 }
 
+/**
+ * Command that looks like
+ * {{{
+ *   DESCRIBE [EXTENDED|FORMATTED] table_name partitionSpec? [AS JSON];
+ * }}}
+ */
+case class DescribeTableJsonCommand(
+   table: TableIdentifier,
+   partitionSpec: TablePartitionSpec,
+   isExtended: Boolean)
+  extends DescribeCommandBase {
+
+  override val output = DescribeCommandSchema.describeTableAttributes(true)
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val result = new ArrayBuffer[Row]
+    val catalog = sparkSession.sessionState.catalog
+
+    if (catalog.isTempView(table)) {
+      if (partitionSpec.nonEmpty) {
+        throw QueryCompilationErrors.descPartitionNotAllowedOnTempView(table.identifier)
+      }
+      val schema = catalog.getTempViewOrPermanentTableMetadata(table).schema
+      describeColsJson(schema, result, header = false)
+    } else {
+      val metadata = catalog.getTableRawMetadata(table)
+      val schema = if (metadata.schema.isEmpty) {
+        // In older versions of Spark,
+        // the table schema can be empty and should be inferred at runtime.
+        sparkSession.table(metadata.identifier).schema
+      } else {
+        metadata.schema
+      }
+
+      addKeyValueToJson(result, "table_name", JString(metadata.identifier.table))
+
+      val catalogName = table.catalog.map(s => s""""$s"""").getOrElse("")
+      val schemaNameOpt = table.database.map(s => s""""$s"""")
+      val namespaceArr = schemaNameOpt.map(schemaName => s"""[$schemaName]""")
+      addKeyValueToJson(result, "catalog_name", parse(catalogName))
```

Comment:
Why do we generate pretty JSON string and then parse it back? Shall we just do `JString(table.catalog.get)`?